Sources/Macros/SampleBuilder/SampleBuilderMacro+Class.swift

```diff
@@ -0,0 +1,200 @@
+/*
+ This source file is part of SwiftAndTipsMacros
+ 
+ Copyright (c) 2023 Pedro Rojas and project authors
+ Licensed under MIT License
+ */
+//
+//  SampleBuilderMacro+Class.swift
+//
+//
+//  Created by Philip Eram on 10/26/23.
+//
+
+import SwiftSyntax
+import DataGenerator
+import DataCategory
+import SwiftSyntaxMacros
+
+extension SampleBuilderMacro {
+    static func SampleBuilderMacroForClass(
+        classDecl: ClassDeclSyntax,
+        numberOfItems: Int,
+        generatorType: DataGeneratorType,
+        context: MacroExpansionContext
+    ) -> [SwiftSyntax.DeclSyntax] {
+        let validParameters = getValidParameterList(
+            from: classDecl,
+            generatorType: generatorType,
+            context: context)
+        
+        let sampleCode = generateSampleCodeSyntax(
+            sampleData: generateSampleData(
+                parameters: validParameters,
+                numberOfItems: numberOfItems,
+                generatorType: generatorType
+            )
+        )
+        
+        return [DeclSyntax(sampleCode)]
+    }
+    
+    static func getValidParameterList(
+        from classDecl: ClassDeclSyntax,
+        generatorType: DataGeneratorType,
+        context: MacroExpansionContext
+    ) -> [ParameterItem] {
+        let storedPropertyMembers = classDecl.memberBlock.members
+            .compactMap {
+                $0.decl.as(VariableDeclSyntax.self)
+            }
+            .filter {
+                $0.isStoredProperty
+            }
+        
+        let initMember = classDecl.memberBlock.members
+            .compactMap {
+                $0.decl.as(InitializerDeclSyntax.self)
+            }
+        
+        if initMember.isEmpty {
+            return storedPropertyMembers.compactMap {
+                if let identifier = $0.bindings.first?
+                    .pattern
+                    .as(IdentifierPatternSyntax.self)?
+                    .identifier.text,
+                   let type = $0.bindings.first?
+                    .typeAnnotation?
+                    .type {
+                    return (identifier, type, getDataCategory(from: $0, generatorType: generatorType, context: context))
+                }
+                
+                return nil
+            }.map {
+                ParameterItem(
+                    identifierName: $0.0,
+                    identifierType: $0.1,
+                    category: $0.2
+                )
+            }
+        }
+        
+        
+        let largestParameterList = initMember
+            .map {
+                getParametersFromInit(initSyntax: $0)
+            }.max {
+                $0.count < $1.count
+            } ?? []
+        
+        return largestParameterList
+    }
+    
+    static func getDataCategory(
```

Comment:
Is this function identical to the one used in Structs, isn't it? If so, Could we remove this one?